duckduckgo_search/cli.py

```diff
@@ -180,13 +180,33 @@ def chat(load, proxy, multiline, timeout, verify, model):
             _save_json(cache_file, cache)
 
 
+class Union(click.ParamType):
+    def __init__(self, types, name="Union"):
+        self.types = types
+        self.name = name
+    
+    def convert(self, value, param, ctx):
+        for type in self.types:
+            try:
+                return type.convert(value, param, ctx)
+            except click.BadParameter:
+                continue
+        
+        self.fail("Parameter is not valid")
+
+
 @cli.command()
 @click.option("-k", "--keywords", required=True, help="text search, keywords for query")
 @click.option("-r", "--region", default="wt-wt", help="wt-wt, us-en, ru-ru, etc. -region https://duckduckgo.com/params")
 @click.option("-s", "--safesearch", default="moderate", type=click.Choice(["on", "moderate", "off"]))
 @click.option("-t", "--timelimit", default=None, type=click.Choice(["d", "w", "m", "y"]), help="day, week, month, year")
 @click.option("-m", "--max_results", default=20, help="maximum number of results, default=20")
-@click.option("-o", "--output", default="print", help="csv, json (save the results to a csv or json file)")
+@click.option("-o", "--output", default="print",
+              help="csv, json (save the results to a csv or json file). optionally specify the full filename",
+              type=Union(
```

Comment:
just define `type=click.Path...`

We don't need Click.Choice here.
If output type is pathlib.Path and output is None, print it as before if output=“print”.